tests/utils/mock_thread_pool.cpp

```diff
@@ -79,11 +79,11 @@ void tieredIndexMock::thread_pool_join() {
     thread_pool.clear();
 }
 
-void tieredIndexMock::thread_pool_wait() {
+void tieredIndexMock::thread_pool_wait(size_t waiting_duration) {
     while (true) {
         if (jobQ.empty() && executions_status.count() == 0) {
```

Comment:
once the queue is empty we need to check only the execution status
also, no need to lock the jobq for this check?

Reply:
Right I guess we should